### PR TITLE
Added react resources

### LINKED ISSUE #2137: [Add] <add React js>
### Site URL

https://react-tutorial.app/

### Description of the above provided site

# React Tutorial app
- It's a website which documents react-js and helps students learn in an interactive way
- It has many modules which help students learn step by step
- The most important feature is that It is interactive
- Where learners need to write and submit the code 

### Category

Frontend

### Sub Category

"reactjs"

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am participating in Hacktoberfest 2023

### PR DESCRIPTION
**Pull request by Srish-ty**
## Fixes Issue 
Closes #2137 

## Changes proposed
- Added resource for React js

### CODE REVIEW

**Comment by Srish-ty** [comment]
@rupali-codes  , Please review the pr and add relevant label.
This contribution is under hacktoberfest.

**Comment by Srish-ty** [comment]
@Anmol-Baranwal @CBID2 , Please review the PR.
And let me know if any changes are expected.
And add relevant labels.

File: database/frontend/reactjs.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[
+  {
+    "name": "ReactJS",
+    "description": "React.js, a widely popular JavaScript library, empowers developers to build dynamic user interfaces with ease. It simplifies the creation of interactive web applications by providing a declarative and component-based approach, making it a go-to choice for developers looking to craft modern and efficient front-end experiences.",
```
Comment (CBID2) [inline database/frontend/reactjs.json:4]:
```suggestion
    "description": "React.js, a widely popular JavaScript library, empowers developers to build dynamic user interfaces with ease. It simplifies the creation of interactive web applications by providing a declarative and component-based approach, making it a go-to choice for developers looking to craft modern and efficient frontend experiences.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Looks good @Srish-ty.Just needs an improvement.

**Comment by Srish-ty** [comment]
> @Srish-ty.There seems to be a deployment failure.

I'd made changes in `data.ts`, probably that's why was getting error.

i've removed that part and now it's deployed successfully. [commit link](https://github.com/rupali-codes/LinksHub/pull/2138/commits/76cc37fcadcf62df8cd4e6185b96abda76af5ed3).

@CBID2 , please review the PR

**Review by CBID2 — APPROVED** [review]
Looks great to me! 😊 @Anmol-Baranwal, you're up! 😊

**Review by Anmol-Baranwal — CHANGES_REQUESTED** [review]
Creating a json file won't create the subcategory.

Follow contributing guidelines to add files.

**Comment by Anmol-Baranwal** [comment]
> Looks great to me! 😊 @Anmol-Baranwal, you're up! 😊

The issue is not assigned 😑 to him

**Comment by Anmol-Baranwal** [comment]
@Srish-ty 
Did you know that if we add a spam label, you won't be able to contribute to Hacktoberfest for a lifetime? This is because participants who are disqualified are not eligible to participate in the upcoming years.

At least comment for assigning the issue, how will someone know.

**Comment by Srish-ty** [comment]
> @Srish-ty Did you know that if we add a spam label, you won't be able to contribute to Hacktoberfest for a lifetime? This is because participants who are disqualified are not eligible to participate in the upcoming years.
> 
> At least comment for assigning the issue, how will someone know.

Sorry sir, but I didn't actually get you. To assign the issue to me? 
Or to add issue to the pr.

**Comment by Anmol-Baranwal** [comment]
> > @Srish-ty Did you know that if we add a spam label, you won't be able to contribute to Hacktoberfest for a lifetime? This is because participants who are disqualified are not eligible to participate in the upcoming years.
> > At least comment for assigning the issue, how will someone know.
> 
> Sorry sir, but I didn't actually get you. To assign the issue to me? Or to add issue to the pr. @Anmol-Baranwal , sorry if this has caused any frustration, but because of GitHub actions the comments in issues are locked and unable to comment 😕

Let me explain the process.

The issues are locked so that people don't tag us unnecessarily to assign the issue. We will always see it at our end and then we will assign the issues and unlock them if we need to clarify something or when we are assigning the issue.

You need to create an issue and allow us to assign you the issue before you start working and submit a PR. This is followed everywhere in Open Source.

This is outlined in the contributing guidelines.
image

You need to follow a predefined flow in order to contribute to Open Source. You may be new, so it is important to understand this. Most of the maintainers will mark it as spam and close it. 

@CBID2 will tell you more about this.

**Comment by Srish-ty** [comment]
> > > @Srish-ty Did you know that if we add a spam label, you won't be able to contribute to Hacktoberfest for a lifetime? This is because participants who are disqualified are not eligible to participate in the upcoming years.
> > > At least comment for assigning the issue, how will someone know.
> > 
> > 
> > Sorry sir, but I didn't actually get you. To assign the issue to me? Or to add issue to the pr. @Anmol-Baranwal , sorry if this has caused any frustration, but because of GitHub actions the comments in issues are locked and unable to comment 😕
> 
> Let me explain the process.
> 
> The issues are locked so that people don't tag us unnecessarily to assign the issue. We will always see it at our end and then we will assign the issues and unlock them if we need to clarify something or when we are assigning the issue.
> 
> You need to create an issue and allow us to assign you the issue before you start working and submit a PR. This is followed everywhere in Open Source.
> 
> This is outlined in the contributing guidelines. image
> 
> You need to follow a predefined flow in order to contribute to Open Source. You may be new, so it is important to understand this. Most of the maintainers will mark it as spam and close it.
> 
> @CBID2 will tell you more about this.

@Anmol-Baranwal , oh okay, sorry for that.
Actually for few projects they asked to not wait till assigned n start working.
But again, sorry.

**Comment by Srish-ty** [comment]
> Creating a json file won't create the subcategory.
> 
> Follow contributing guidelines to add files.
Yeah in contributions.md , there are three types mentioned.
And I wanted to work on adding links/ resources part

**Comment by Anmol-Baranwal** [comment]
> @Anmol-Baranwal , oh okay, sorry for that. Actually for few projects they asked to not wait till assigned n start working. But again, sorry.

It's not your fault; this process is followed everywhere, from large organizational repositories (like Microsoft's) to smaller ones. Projects that don't follow this can lead to chaos. What if someone else makes a PR for the same issue, or if the PR doesn't align with the codebase, it may be closed, and that will reflect in your profile.

As I mentioned, maintainers of decent repositories will mark such PRs as spam. No worries, just make sure to read the contributing guidelines because it will always be mentioned.

**Comment by CBID2** [comment]
> > @Anmol-Baranwal , oh okay, sorry for that. Actually for few projects they asked to not wait till assigned n start working. But again, sorry.
> 
> It's not your fault; this process is followed everywhere, from large organizational repositories (like Microsoft's) to smaller ones. Projects that don't follow this can lead to chaos. What if someone else makes a PR for the same issue, or if the PR doesn't align with the codebase, it may be closed, and that will reflect in your profile.
> 
> As I mentioned, maintainers of decent repositories will mark such PRs as spam. No worries, just make sure to read the contributing guidelines because it will always be mentioned.

@Srish-ty's issue has been approved and assigned @Anmol-Baranwal. Now you can review it! :)

**Review by Anmol-Baranwal — CHANGES_REQUESTED** [review]
if you're creating a subcategory, then you need to update data.ts and index.ts as well.

It won't be created just by adding a json file. Follow contributing guidelines.

**Comment by Anmol-Baranwal** [comment]
@Srish-ty 
I'm closing this issue due to inactivity. Please feel free to reopen it if you'd like to contribute or have further discussions.

**Pull request closed without merging**